contracts/defi/provider/CompoundV2Provider.sol

```diff
@@ -0,0 +1,405 @@
+        address _token, 
+        uint256 _amount, 
+        uint256 _period, 
+        address[] calldata _oracles
+    ) 
+        external 
+        returns (uint256 _invested)
+    {
+        require(_oracles.length == 2, "CompoundV2: invalid oracles length");
+        address cToken = CompoundRegistry(_oracles[1]).getCToken(_token);
+        mint(_wallet, cToken, _token, _amount);
+        _invested = _amount;
+    }
+
+    /**
+     * @dev Exit invested postions.
+     * @param _wallet The target wallet.s
+     * @param _token The token address.
+     * @param _fraction The fraction of invested tokens to exit in per 10000. 
+     * @param _oracles (optional) The address of one or more oracles contracts that may be used by the provider to query information on-chain.
+     */
+    function removeInvestment(
+        BaseWallet _wallet, 
+        address _token, 
+        uint256 _fraction, 
+        address[] calldata _oracles
+    ) 
+        external 
+    {
+        require(_oracles.length == 2, "CompoundV2: invalid oracles length");
+        require(_fraction <= 10000, "CompoundV2: invalid fraction value");
+        address cToken = CompoundRegistry(_oracles[1]).getCToken(_token);
+        uint shares = CToken(cToken).balanceOf(address(_wallet));
+        redeem(_wallet, cToken, shares.mul(_fraction).div(10000));
+    }
+
+    /**
+     * @dev Get the amount of investment in a given token.
+     * @param _wallet The target wallet.
+     * @param _token The token address.
+     * @param _oracles (optional) The address of one or more oracles contracts that may be used by the provider to query information on-chain.
+     * @return The value in tokens of the investment (including interests) and the time at which the investment can be removed.
+     */
+    function getInvestment(
+        BaseWallet _wallet, 
+        address _token, 
+        address[] calldata _oracles
+    ) 
+        external 
+        view
+        returns (uint256 _tokenValue, uint256 _periodEnd) 
+    {
+        require(_oracles.length == 2, "CompoundV2: invalid oracles length");
+        address cToken = CompoundRegistry(_oracles[1]).getCToken(_token);
+        uint amount = CToken(cToken).balanceOf(address(_wallet));
+        uint exchangeRateMantissa = CToken(cToken).exchangeRateStored();
+        _tokenValue = amount.mul(exchangeRateMantissa).div(10 ** 18);
+        _periodEnd = 0;
+    }
+
+    /* ********************************** Implementation of Loan ************************************* */
+
+    /**
+     * @dev Opens a collateralized loan.
+     * @param _wallet The target wallet.
+     * @param _collateral The token used as a collateral.
+     * @param _collateralAmount The amount of collateral token provided.
+     * @param _debtToken The token borrowed.
+     * @param _debtAmount The amount of tokens borrowed.
+     * @param _oracles (optional) The address of one or more oracles contracts that may be used by the provider to query information on-chain.
```

Comment:
I would specify here that `_oracles[0]` should be the Comptroller and `_oracles[1]` should be the Compound Registry

Reply:
Done.